besu/src/main/java/org/hyperledger/besu/controller/CliqueBesuControllerBuilder.java

```diff
@@ -132,14 +134,18 @@ protected PluginServiceFactory createAdditionalPluginServices(final Blockchain b
   @Override
   protected CliqueContext createConsensusContext(
       final Blockchain blockchain, final WorldStateArchive worldStateArchive) {
-    return new CliqueContext(
-        new VoteTallyCache(
-            blockchain,
-            new VoteTallyUpdater(epochManager, blockInterface),
+    final CliqueContext cliqueContext =
+        new CliqueContext(
+            new VoteTallyCache(
+                blockchain,
+                new VoteTallyUpdater(epochManager, blockInterface),
+                epochManager,
+                blockInterface),
+            new VoteProposer(),
             epochManager,
-            blockInterface),
-        new VoteProposer(),
-        epochManager,
-        blockInterface);
+            blockInterface);
+    installCliqueBlockChoiceRule(blockchain, cliqueContext);
```

Comment:
Started going down a rabbit hole to remove this mutation. I didn't see a way out that isn't too intrusive.

Reply:
The only way to be less intrusive is for a more invasive DI change, like going to Dagger.